src/engine/config.rs

```diff
@@ -324,6 +324,38 @@ impl Config {
     }
 }
 
+fn expand_path<S: AsRef<str>>(input: S) -> PathBuf {
+    let input = input.as_ref();
+
+    // Expand tilde ~
+    let input = if let Some(stripped) = input.strip_prefix("~/") {
+        if let Some(home) = dirs::home_dir() {
+            home.join(stripped)
+        } else {
+            PathBuf::from(input)
+        }
+    } else {
+        PathBuf::from(input)
+    };
+
+    // Convert to lossy string once
+    let input_str = input.to_string_lossy();
+
+    // Clone needed here to avoid use-after-move conflict
+    let with_expanded_vars = shellexpand::env(input_str.as_ref())
+        .unwrap_or_else(|_| input_str.clone().into_owned().into());
+
+    Path::new(&*with_expanded_vars).to_path_buf()
```

Comment:
Why do we need to do the ref-deref dance here?

Reply:
Because I don't know what I'm doing and just try to please the linter xD

Good catch! Fixed now